torch/_dynamo/decorators.py

```diff
@@ -1,3 +1,4 @@
+# ruff: noqa: TCH004
```

Comment:
This file has some weird dynamic imports in the else statement of TYPE_CHECKING.